Sources/FeatherDatabase/Database/DatabaseGroupFilterInterface.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  File.swift
+//
+//
+//  Created by Tibor Bodecs on 11/03/2024.
+//
+
+import SQLKit
+
+public protocol DatabaseGroupFilterInterface {
+
+    var relation: DatabaseFilterRelation { get }
+    var columns: [any DatabaseFilterInterface] { get }
+}
+
+//// SELECT * FROM galaxies WHERE name != NULL AND (name == ? OR name == ?)
+//_ = try await self.db.select()
+//    .column("*")
+//    .from("galaxies")
+//    .where("name", .notEqual, SQLLiteral.null)
+//    .where { $0
+//        .orWhere("name", .equal, SQLBind("Milky Way"))
+//        .orWhere("name", .equal, SQLBind("Andromeda"))
+//    }
+//    .all()
+
+extension SQLSelectBuilder {
+
+    func applyGroupFilters<T: DatabaseGroupFilterInterface>(
+        _ groupFilters: [T]
+    ) -> Self {
+        var res = self
+        for group in groupFilters {
+            res = res.where { p in
+                var p = p
+                for filter in group.columns {
+                    switch group.relation {
+                    case .and:
+                        p = p.where(
+                            filter.column.sqlValue,
+                            filter.operator,
+                            filter.value
+                        )
+                    case .or:
+                        p = p.orWhere(
+                            filter.column.sqlValue,
+                            filter.operator,
+                            filter.value
+                        )
+                    }
+                }
+                return p
+            }
+        }
+        return res
+    }
+}
+
+extension SQLSelectBuilder {
+
+    func applyFilter(
```

Comment:
In practice it is an API guarantee that you do _not_ have to actually keep track of the return values of `Self`-returning methods on any builder (with the sole exception of a `.returning()` clause, which is a special case due to legacy design constraints); as such, you can significantly simplify this method.

Reply:
It's good to know this, will improve the result builders in a later version.